File: rei-do-chopp-site/src/app/restockings/models/requests/restocking-request.ts
import { RestockingProductRequest } from './restocking-product-request';
import { RestockingAdditionalFeeRequest } from './restocking-additional-fee-request';

export class RestockingRequest {
  Date: Date;
  Discount?: number;
  RestockingProducts: RestockingProductRequest[];
  RestockingAdditionalFees: RestockingAdditionalFeeRequest[];

  constructor(params: Partial<RestockingRequest>) {
    this.Date = params.Date || null;
    this.Discount = params.Discount || null;
    this.RestockingProducts = params.RestockingProducts || [];
    this.RestockingAdditionalFees = params.RestockingAdditionalFees || [];
  }
}

File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Restockings/Requests/RestockingRequest.cs
using ReiDoChopp.DataTransfer.RestockingAdditionalFees.Requests;
using ReiDoChopp.DataTransfer.RestockingProducts.Requests;

namespace ReiDoChopp.DataTransfer.Restockings.Request
{
    public class RestockingRequest
    {
        public DateTime Date { get; set; }
        public double? Discount { get; set; }
        public RestockingProductRequest[] RestockingProducts { get; set; }
        public RestockingAdditionalFeeRequest[] RestockingAdditionalFees { get; set; }
    }
}
